docs/examples/ODE_example.py

```diff
@@ -0,0 +1,367 @@
+        d_acc = lambda a:function(a, timestamp=timestamp)
+
+        # Initialise the matrix
+        A = np.zeros([nx, nx])
+        istate = [i for i in state]
+
+        # use autograd to create the matrix components
+        jac_rows = torch.autograd.functional.jacobian(d_acc, torch.tensor(istate))
+        for i, r in enumerate(jac_rows):
+            A[i] = r
+
+        return A
+
+
+    def _do_linearise(self, da, state, dt):
+        """ Function that linearises the ODE.
+
+            da : function = ODE to be linearised
+            state : StateVector/np.array = state where differentiate the ODE
+            dt : timedelta = time interval where differentiate the ODE
+        """
+
+        # Define the timestamp and dimension
+        timestamp = state.timestamp
+        nx = self.ndim_state
+
+        # Obtain the jacobian of the ODE
+        dA = self._get_jacobian(da, state, timestamp=timestamp)
+        # Make the jacobian available for all the class at specific time and state
+        self.dA = dA
+
+        # Compute the integral of the Jacobian
+        # Get \int e^{dA*s}\,ds
+        int_eA = expm(dt * np.block([
+            [dA, np.identity(nx)],
+            [np.zeros([nx, 2 * nx])
+             ]]))[:nx, nx:]
+
+        # Get new value of x
+        x = [i for i in state.state_vector]
+        newx = x + int_eA @ da(torch.tensor(x))
+
+        return newx
+
+
+    def jacobian(self, state, **kwargs):
+        """ Function for the jacobian"""
+
+        da = self.differential_equation
+        timestamp = state.timestamp
+        dt = kwargs['time_interval'].total_seconds()
+
+        dA = self._get_jacobian(da, state, timestamp=timestamp)
+        # Share the jacobian across the class, might not be needed
+        self.dA = dA
+        A_d = expm(dA * dt)
+
+        return A_d
+
+
+    def function(self, state, noise=False, **kwargs) -> StateVector:
+        """ Transition function that uses the linearised function """
+
+        # use the differential equation and time delta
+        da = self.differential_equation
+        dt = kwargs['time_interval'].total_seconds()
+        # create a new state using the linearised ODE
+        new_state = self._do_linearise(da, state, dt)
+
+        if isinstance(noise, bool) or noise is None:
+            if noise:
+                noise = self.rvs(prior=state, **kwargs)
+            else:
+                noise = 0
+
+        return StateVector(new_state) + noise
+
+
+    def covar(self, time_interval, **kwargs):
+        """ Function to create the covariance matrix
+
+            We use the jacobian obtained previously using
+            self.dA
+        """
+
+        # define the dimension and timedelta
+        nx = self.ndim_state
+        dt = time_interval.total_seconds()
+
+        # Get Q
+        q_xdot, q_ydot, q_zdot = self.linear_noise_coeffs
+        dQ = np.diag([0., q_xdot, 0., q_ydot, 0., q_zdot])
+
+        G = expm(dt * np.block([[-self.dA, dQ],
```

Comment:
Not ideal having `self.dA` here as it may not be set. But can see issue in that you need the `state` here in order to calculate it. Not sure what solution would be for this...